### ISSUE TITLE
No function despite imagick

### ISSUE THREAD

**Issue by mariusnop** [cowai/camerarawpreviews#20]
Hello,

and first of all thanks for this great app!

I have the problem that the app does not work for me. 
The current situation:

- Freenas FreeBSD as operating system
- Nextcloud 13 installed (works without problems)
- php7.2 incl imagick installed, this also runs "php -m".

but it is not possible to see RAW data in the preview.

**Comment by cowai** [comment]
@mariusnop You probably do not have perl installed right? I bundle a static linux perl binary, but it's only used if there is no perl in PATH. I am planning on making it only opt-in for getting the static perl binary though.
If you already have perl installed, then maybe you are using a raw file that is not supported. Can you upload a sample so I can check?

**Comment by mariusnop** [comment]
I hadn't installed pearl, but I did that later, without success. The files still cannot be opened. I use different RAW files from rawsamples.ch .

**Comment by cowai** [comment]
can you try
````bash
perl $APPFOLDER/camerarawpreviews/vendor/jmoati/exiftool-bin/exiftool -json -preview:all SAMPLE.CR2"

**Comment by mariusnop** [comment]
SAMPLE.CR2 cant be found

Do I have to upload a CR2 file or should it already exist as an example?

**Comment by cowai** [comment]
Yes that seems right.
I have experienced nextcloud caching corrupted previews. can you try uploading SAMPLE.CR2 with a new name into nextcloud web ui and see if it works? Also clear logs before it and then try to see if you get a new error into nextcloud.

**Comment by mariusnop** [comment]
`message":"Exception: Preview generation has failed`

**Comment by cowai** [comment]
I am afraid this must be some freebsd-specific issue. I have no experience with it. If you could reproduce it in an environment I could get access to, I might be able to tweak the code to make it work. If you could make a vm image or give ssh access to a test environment that would work.

**Comment by mariusnop** [comment]
You are welcome to have current data, as this is a test system. I document every step so that I can follow the installation step by step.

**Comment by cowai** [comment]
@mariusnop 
The fix is now in latest version 0.5.5. Both in Nextcloud Apps and ownCloud Marketplace.

**Comment by mariusnop** [comment]
Thanks for your work, I'm really excited about your support and how quickly you can correct your app for bugs. Everything works perfectly.

Thank you!
